tsdb/engine/tsm1/compact.go

```diff
@@ -77,10 +76,10 @@ type tsmGeneration struct {
 }
 
 // size returns the total size of the generation
-func (t *tsmGeneration) size() int64 {
-	var n int64
+func (t *tsmGeneration) size() uint32 {
```

Comment:
all sizes have to be in the `uint32` range because that's what the offsets are in the data files. Before there was a mix of `int64` and `int` which was causing failures on 32 bit systems if the max size was set to higher than a signed int32 could handle.

This is more explicit and matches up with the actual limitations of the file format.